test/unit/region.spec.js

```diff
@@ -337,6 +337,22 @@ describe('region', function() {
               expect(this.$parentEl).to.contain.$html('<div id="region"></div>');
             });
           });
+
+          describe('and restore, when view was self destroyed', function() {
+            beforeEach(function() {
+              this.view.trigger('before:destroy', this.view);
+              this.view._removeElement();
```

Comment:
I think `this.view.destroy();` should do the event emitting instead in order to match a real scenario.